translator/internaldata/traces_to_oc.go

```diff
@@ -35,29 +35,44 @@ var (
 )
 
 func TraceDataToOC(td data.TraceData) []consumerdata.TraceData {
-	ocTraceData := consumerdata.TraceData{
-		SourceFormat: sourceFormat,
-	}
+	resourceSpans := td.ResourceSpans()
 
-	resourceSpansList := td.ResourceSpans()
+	if resourceSpans.Len() == 0 {
+		return nil
+	}
 
-	ocResourceSpansList := make([]consumerdata.TraceData, 0, len(resourceSpansList))
+	ocResourceSpansList := make([]consumerdata.TraceData, 0, resourceSpans.Len())
 
-	for _, resourceSpans := range resourceSpansList {
-		ocTraceData.Node, ocTraceData.Resource = internalResourceToOC(resourceSpans.Resource())
-		ocSpans := make([]*octrace.Span, 0, td.SpanCount())
-		for _, il := range resourceSpans.InstrumentationLibrarySpans() {
-			for _, span := range il.Spans() {
-				ocSpans = append(ocSpans, spanToOC(span))
-			}
-		}
-		ocTraceData.Spans = ocSpans
-		ocResourceSpansList = append(ocResourceSpansList, ocTraceData)
+	for i := 0; i < resourceSpans.Len(); i++ {
+		ocResourceSpansList = append(ocResourceSpansList, ResourceSpansToOC(resourceSpans.Get(i)))
 	}
 
 	return ocResourceSpansList
 }
 
+func ResourceSpansToOC(rs data.ResourceSpans) consumerdata.TraceData {
+	ocTraceData := consumerdata.TraceData{
+		SourceFormat: sourceFormat,
+	}
+	ocTraceData.Node, ocTraceData.Resource = internalResourceToOC(rs.Resource())
+	ilss := rs.InstrumentationLibrarySpans()
+	if ilss.Len() == 0 {
+		return ocTraceData
+	}
+	// Approximate the number of the metrics as the number of the metrics in the first
+	// instrumentation library info.
+	ocSpans := make([]*octrace.Span, 0, ilss.Get(0).Spans().Len())
+	for i := 0; i < ilss.Len(); i++ {
+		// TODO: Handle instrumentation library name and version.
+		spans := ilss.Get(0).Spans()
```

Comment:
```suggestion
		spans := ilss.Get(i).Spans()
```
Maybe add a test that catches this.

Reply:
Done. Will add tests after this PR gets merged, this changes a lot of files and is hard to keep changing it.